compiler/src/main/java/dagger/internal/codegen/ComponentDescriptor.java

```diff
@@ -410,9 +412,27 @@ private ComponentDescriptor create(
       for (TypeElement componentDependency : componentDependencyTypes) {
         List<ExecutableElement> dependencyMethods =
             ElementFilter.methodsIn(elements.getAllMembers(componentDependency));
+        HashMultimap<QualifiedMethodNameAndType, Equivalence.Wrapper<TypeMirror>> componentMethodType
```

Comment:
It looks like you're trying to dedup methods based on signature.  Can you not use `MethodSignature` to do that rather than make a new type?